### PR TITLE
Initialize devcontainer

### PR DESCRIPTION
**Pull request by cathalnoonan**
Adds a `devcontainer` for running the code in the browser, or in a pre-configured environment within Visual Studio Code.

The container image includes .NET 6.0 and Node JS (v18, the current LTS).

Also addresses the following issues:
- `npm install` didn’t install the global packages. Instead `npm-run-all` is installed as a devDependency.
- Commands in the package.json relied on Windows specifics (`npm run clean` and `npm run generate-toc`).
- Packages in the lib folder were Windows specific. These should be consumed as their `dotnet tool` counterparts to avoid Windows runtime dependencies.

---

As a minimum to get the devcontainer working properly, the following commands should be executed **manually**:
- ~npm install~ (now run automatically when the devcontainer starts up)
- ~npm run setup~ (now run automatically when the devcontainer starts up)
- npm run build
- npm run preview

### CODE REVIEW

**Comment by cathalnoonan** [comment]
image

**Comment by cathalnoonan** [comment]
The dependencies included in the `lib` folder were pushed as dotnet tools. I've removed the `lib` folder and its contents as it's no longer needed.

**Review by ryanspain — APPROVED** [review]
Thank you @cathalnoonan!

File: .config/dotnet-tools.json
Comment (ryanspain) [inline .config/dotnet-tools.json:1]: Removed unused docfx companion tools.

**Review by ryanspain — COMMENTED** [review]
Merged without unused docfx companion tools.

**Pull request merged**